storage/src/merkle_tree_db_adapter.cpp

```diff
@@ -328,25 +342,30 @@ Status DBAdapter::getBlockById(BlockId blockId, Sliver &block) const {
   for (const auto &[key, keyData] : blockNode.keys) {
     if (!keyData.deleted) {
       Sliver value;
-      if (const auto status = db_->get(DBKeyManipulator::genDataDbKey(key, blockId), value); !status.isOK()) {
+      if (const auto status = db_->get(DBKeyManipulator::genDataDbKey(key, blockNode.stateRootVersion.value()), value);
```

Comment:
Is this because the blockId no longer matches the root version ?

Reply:
Yes.